mkl_umath/src/mkl_umath_loops.c.src

```diff
@@ -2659,44 +2656,47 @@ mkl_umath_@TYPE@__ones_like(char **args, const npy_intp *dimensions, const npy_i
     }
 }
 
-/* TODO: USE MKL */
 void
 mkl_umath_@TYPE@_conjugate(char **args, const npy_intp *dimensions, const npy_intp *steps, void *NPY_UNUSED(func)) {
-    UNARY_LOOP {
-        const @ftype@ in1r = ((@ftype@ *)ip1)[0];
-        const @ftype@ in1i = ((@ftype@ *)ip1)[1];
-        ((@ftype@ *)op1)[0] = in1r;
-        ((@ftype@ *)op1)[1] = -in1i;
-    }
+    const int contig = IS_UNARY_CONT(@type@, @type@);
+    const int disjoint_or_same = DISJOINT_OR_SAME(args[0], args[1], dimensions[0], sizeof(@type@));
+    const int can_vectorize = contig && disjoint_or_same;
+
+    if(can_vectorize && dimensions[0] > VML_TRANSCEDENTAL_THRESHOLD) {
+        CHUNKED_VML_CALL2(v@s@Conj, dimensions[0], @type@, args[0], args[1]);
+        /* v@s@Conj(dimensions[0], (@type@*) args[0], (@type@*) args[1]); */
+    } else {
+		UNARY_LOOP {
+			const @ftype@ in1r = ((@ftype@ *)ip1)[0];
+			const @ftype@ in1i = ((@ftype@ *)ip1)[1];
+			((@ftype@ *)op1)[0] = in1r;
+			((@ftype@ *)op1)[1] = -in1i;
+		}
+	}
 }
 
-/* TODO: USE MKL */
 void
 mkl_umath_@TYPE@_absolute(char **args, const npy_intp *dimensions, const npy_intp *steps, void *NPY_UNUSED(func))
 {
+    const int contig = IS_UNARY_CONT(@type@, @ftype@);
+    const int disjoint_or_same = DISJOINT_OR_SAME_TWO_DTYPES(args[0], args[1], dimensions[0], sizeof(@type@), sizeof(@ftype@));
+    const int can_vectorize = contig && disjoint_or_same;
     int ignore_fpstatus = 0;
-
-    // FIXME: abs function VML for complex numbers breaks FFT test_basic.py
-    //if(steps[0]/2 == sizeof(@ftype@) && steps[1] == sizeof(@ftype@) && dimensions[0] > VML_TRANSCEDENTAL_THRESHOLD) {
-#if @SUPPORTED_BY_VML@
-    if(0 == 1) {
+    
+    if(can_vectorize && dimensions[0] > VML_TRANSCEDENTAL_THRESHOLD) {
         ignore_fpstatus = 1;
-        CHUNKED_VML_CALL2(v@s@Abs, dimensions[0], @ftype@, args[0], args[1]);
-        /* v@s@Abs(dimensions[0], (@ftype@ *) args[0], (@ftype@ *) args[1]); */
-    } else
-#endif
-    {
+        CHUNKED_VML_CALL2(v@s@Abs, dimensions[0], @type@, args[0], args[1]);
+        /* v@s@Abs(dimensions[0], (@type@*) args[0], (@type@*) args[1]); */
+    } else {
         UNARY_LOOP {
             const @ftype@ in1r = ((@ftype@ *)ip1)[0];
             const @ftype@ in1i = ((@ftype@ *)ip1)[1];
-            if(in1r == 0.0 && in1i == 0.0){
-                ignore_fpstatus = 1;
-            }
+            ignore_fpstatus = npy_isnan(in1r) && npy_isnan(in1i);
             *((@ftype@ *)op1) = hypot@c@(in1r, in1i);
```

Comment:
I wonder if we would like to use `std::hypot` here or `npy_hypot` here?.. and what is the intention

Reply:
Yep, I meant [hypot](https://en.cppreference.com/w/c/numeric/math/hypot) from c99, not `std::hypot`.

I'm not quite sure if `npy_hypot` would be better to use here. I was just curious what the reason was for not using it.
Now it's crystal that it requires static linking with `libnpymath.a`.

Probably we can left that out of that PR and to keep the previous behavior as for now.
If we are going to use NumPy API here, it would probably make sense to do it for every place where possible. But it would make sense to investigate and decide separately, in the dedicated PR.